Bio/SeqUtils/__init__.py

```diff
@@ -667,6 +667,46 @@ def calculate(self, sequence):
 
         return exp(cai_value / cai_length)
 
+    def optimize(self, sequence, seq_type="DNA", table=standard_dna_table):
+        """Return a new DNA sequence with preferred codons only."""
+        try:  # If seq record is provided, convert to sequence
+            sequence = sequence.seq
+        except AttributeError:  # not a  SeqRecord object
+            pass
+        seq = sequence.upper()
+        # Make dict with amino acids referencing preferred codons
+        pref_codons = {aminoacid: [] for aminoacid in table.protein_alphabet}
```

Comment:
Using an empty list as a placeholder for unset items is odd. How about starting with an empty dictionary? And then in line 681 use `if aminoacid in pref_codons:` instead of `if pref_codons[aminoacid] != []:`

Reply:
Right, that is more natural, I have updated as suggested. Thanks!